torch/csrc/autograd/engine_without_python.cpp

```diff
@@ -0,0 +1,38 @@
+#include "torch/csrc/autograd/engine_without_python.h"
+
+#include "torch/csrc/autograd/engine.h"
+#include "torch/csrc/autograd/function.h"
+#include "torch/csrc/autograd/functions/basic_ops.h"
+#include "torch/csrc/autograd/grad_mode.h"
+#include "torch/csrc/autograd/variable.h"
+#include "torch/csrc/utils/auto_gpu.h"
+
+#include <atomic>
+#include <condition_variable>
+#include <cstdint>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <mutex>
+#include <set>
+#include <string>
+#include <thread>
+#include <unordered_set>
+#include <typeinfo>
+#include <sstream>
+#include <queue>
+#include <TH/TH.h>
+
+#ifdef WITH_CUDA
+#include <cuda.h>
+#include <THC/THC.h>
+#endif
+namespace torch{ namespace autograd{
+    
+Engine& Engine::getDefaultEngine(){
```

Comment:
We need to have only 1 definition of getDefaultEngine but this will introduce 2. The other one is in python_engine.cpp. Instead, we should define getDefaultEngine in Engine.cpp and replace uses of `engine` inside `python_engine.cpp` with calls to that function.

Reply:
Yes, and that is why I deleted the definition of getDefaultEngine in python_engine.cpp. The originally logic is like this: if there is no-python, then define getDefaultEngine in engine.cpp; else do nothing, and use the (same) definition of getDefaultEngine in python_engine.cpp. Since these two definitions are same, I think @apaszke is right that we can simply move this piece of code into engine.cpp.